src/Model/EndpointLocator.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+namespace Muffin\Webservice\Model;
+
+use Cake\Core\App;
+use Cake\Datasource\ConnectionManager;
+use Cake\Utility\Inflector;
+use RuntimeException;
+
+/**
+ * Class EndpointLocator
+ *
+ * Should implement the LocatorInterface
+ * @see \Cake\ORM\Locator\LocatorInterface
+ * @see https://github.com/cakephp/cakephp/issues/12014
+ */
+class EndpointLocator
+{
+    /**
+     * Configuration for aliases.
+     *
+     * @var array
+     */
+    protected $_config = [];
+
+    /**
+     * Instances that belong to the registry.
+     *
+     * @var \Muffin\Webservice\Model\Endpoint[]
+     */
+    protected $_instances = [];
+
+    /**
+     * Contains a list of options that were passed to get() method.
+     *
+     * @var array
+     */
+    protected $_options = [];
+
+    /**
+     * Set an Endpoint instance to the registry
+     *
+     * @param string $alias The alias to set.
+     * @param \Muffin\Webservice\Model\Endpoint $object The table to set.
+     * @return \Muffin\Webservice\Model\Endpoint
+     */
+    public function set($alias, Endpoint $object)
```

Comment:
If/when the `LocatorInterface` get's updated the 2nd parameter will very likely be type hinted as `RepositoryInterface` so if we use `Endpoint` here the class will still be not implement the interface.

Let's see what response we get from the team on that issue.